pulse2percept/tests/test_effectivecurrent2brightness.py

```diff
@@ -1,35 +1,59 @@
-import tempfile
 import numpy as np
 import numpy.testing as npt
 import pulse2percept.electrode2currentmap as e2cm
 import pulse2percept.effectivecurrent2brightness as ec2b
+from pulse2percept.utils import TimeSeries
+
+
+def test_pulse2percept():
+    # All ways to break the function call
+    retina = e2cm.Retina(xlo=0, xhi=0, ylo=0, yhi=0)
+    implant = e2cm.ElectrodeArray('epiretinal', 0, 0, 0, 0)
+    electrode = e2cm.Electrode('epiretinal', 0, 0, 0, 0)
+    pt = TimeSeries(0.001, np.ones(1000))
+    tm = ec2b.TemporalModel()
+
+    # Ways to break the function call (not including invalid pulse
+    # trains, which are tested further down)
+    fl_dummy = 10.2
+    engine = 'serial'
+    dojit = False
+    npt.assert_raises(TypeError, ec2b.pulse2percept, pt, implant, retina,
+                      fl_dummy, fl_dummy, fl_dummy, True, engine, dojit)
+    npt.assert_raises(TypeError, ec2b.pulse2percept, pt, electrode,
+                      tm, retina, fl_dummy, fl_dummy, fl_dummy, True,
+                      engine, dojit)
+    npt.assert_raises(TypeError, ec2b.pulse2percept, pt, implant,
+                      retina, retina, fl_dummy, fl_dummy, fl_dummy,
+                      True, engine, dojit)
+
+    # Smoke testing
+    ec2b.pulse2percept(pt, implant, tm, retina, engine=engine, dojit=dojit)
 
 
 def test_brightness_movie():
-    retina_file = tempfile.NamedTemporaryFile().name
     sampling = 1
     xlo = -2
     xhi = 2
     ylo = -3
     yhi = 3
     retina = e2cm.Retina(xlo=xlo, xhi=xhi, ylo=ylo, yhi=yhi,
-                         sampling=sampling, axon_map=retina_file)
+                         sampling=sampling, loadpath='')
```

Comment:
I could try to argue about how unlikely that is with the name filename generator, or start messing with tempdirs... :) But a much easier solution seemed to be the introduction of an optional input argument `save_data` that is set to True by default but gets set to `False` in all tests. Hence no new files will be generated in tests.